app/models/energy/ScheduleFile.py

```diff
@@ -0,0 +1,57 @@
+"""Schedule File Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+
+
+class UnitType (str, Enum):
+    dimensionless: 'Dimensionless'
+    temperature: 'Temperature'
+    delta_temperature: 'DeltaTemperature'
+    precipitation_rate: 'PrecipitationRate'
+    angle: 'Angle'
+    convection_coefficient: 'ConvectionCoefficient'
+    activity_level: 'ActivityLevel'
+    velocity: 'Velocity'
+    capacity: 'Capacity'
+    power: 'Power'
+    availability: 'Availability'
+    percent: 'Percent'
+    control: 'Control'
+    mode: 'Mode'
+
+
+class ScheduleFile(BaseModel):
+    """A list of values used to create a .csv file for the entire year or a period of
+  analysis within the year."""
+
+    type: Enum('ScheduleFile', {'type': 'ScheduleFile'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    start_day_of_year: int = Schema(
+        ...,
+        ge=1,
+        le=366,
+        description='Used to define the start day for the analysis period.'
+    )
+
+    values: List[int] = Schema(
+        ...,
+        minItems=8760,
```

Comment:
I believe that the min here should be 24 since, if someone makes a schedule with start day of 365 on a year that is not a leap year, the number of values in the list should be 24 for the single day of the year.

Reply:
That makes sense!

Reply:
There are some limitations in E+.  Approach 1 is to accept CSV from user and then generate a compliant CSV to pass to E+.  Approach 2 is to limit your users to valid E+ CSVs.

https://github.com/NREL/EnergyPlus/issues/4154
https://github.com/NREL/EnergyPlus/issues/6898

Reply:
@macumber ,
I had some discussions with @mostaphaRoudsari about this and we are in favor of something closer to approach 1, where we have the user send a start day of the year and the minimum number of values needed to define the schedule for the simulation RunPeriod. These two parameters can be used to create the E+ compliant CSV by simply adding filler values for the unimportant hours.  Also, we are sure that we would rather have the user specify the schedule values within the JSON and have the energy_model_measure generate the compliant CSVs rather than have users send separate CSV files that we then have to validate in another script.